ui/nuxeo-filters-behavior.js

```diff
@@ -273,4 +273,23 @@ export const FiltersBehavior = {
     const processes = document.contextParameters.runnableWorkflows;
     return processes && processes.length > 0;
   },
+
+  /**
+   * Checks if the given document or its main content, can be deleted by the given user
+   */
+  canDelete(document, user) {
+    const underRetentionOrLegalHold = this.isUnderRetentionOrLegalHold(document);
+    const hasRemovePermission = this.hasPermission(document, 'Remove');
+    if (!underRetentionOrLegalHold && hasRemovePermission) {
+      return true;
+    }
+    return (
+      (this.hasAdministrationPermissions(user) || this.hasPermission(document, 'RemoveRecords')) &&
+      Nuxeo &&
+      Nuxeo.UI &&
+      Nuxeo.UI.config &&
+      Nuxeo.UI.config.retention &&
+      String(Nuxeo.UI.config.retention.mode) === 'governance'
```

Comment:
so if `Nuxeo.UI.config.retention.mode` is not set we can't delete documents ?
also, shouldn't user permissions on the document be enough to do this checks? the ideal of these hardcoded client side checks is for them to mimic what the server allows you to do, if toggling a client side flag prevents me from deleting a document but the REST API still allows it then this is a false sense of security so it should go in a filter in a UI contribution at most.

Reply:
Hey @nelsonsilva thanks for the feedbacks, I was going to ask you if we can meet. I want to share with you these kind of needs/requirements. I have opened a PR on backend part for almost the same rules. on frontend we want to implement the same things to hide or show the button(s). I asked myself how we can improve these double check, how we can do it easily; contribution, inheritance of elements ... I'll to send an invitation this week if you agree to talk about it.